File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& a) {
        
        int i=0,j=0,k=a.size()-1;
        
        while(j<=k)
        {
            if(a[j] == 0)
                swap(a[i++],a[j++]);
            
            else if(a[j]==1)
                j++;
            else swap(a[j],a[k--]);
        }
        
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    void swap(int[] nums,int i,int j)
    {
       int temp = nums[i];
       nums[i] = nums[j];
       nums[j] = temp;
    }
    public void sortColors(int[] nums) 
    {
        int low = 0,n=nums.length;
        int high = n-1;
        int mid=0;
        
        while(mid<=high)
        {
            if(nums[mid] == 2) 
            {
                swap(nums,mid,high);
                high--;
            }
             else if(nums[mid] == 1) 
            {
               mid++;
            }
            else if(nums[mid] == 0 )
            {
                swap(nums,mid,low);
                low++;
                mid++;
            }
           
          
        }
        
    }
}